### PR TITLE
Add a ALTER operation to change distribution type from RANDOM to HASH

### PR DESCRIPTION
**Pull request by morningman**
Random distribution is no longer supported since version 0.9.
And we need a way to convert the random distribution to hash distribution.
    
`ALTER TABLE db.tbl SET ("distribution_type" = "hash");`

### CODE REVIEW

File: docs/documentation/cn/sql-reference/sql-statements/Data Definition/ALTER TABLE.md [hunk 1]
```diff
@@ -232,7 +232,12 @@
         PROPERTIES ("bloom_filter_columns"="k1,k2,k3");
 
     12. 修改表的Colocate 属性
-        ALTER TABLE example_db.my_table set ("colocate_with"="t1");
+
+        ALTER TABLE example_db.my_table set ("colocate_with" = "t1");
+
+    13. 将表的分桶方式由 Random Distribution 改为 Hash Distribution
```
Comment (imay) [inline docs/documentation/cn/sql-reference/sql-statements/Data Definition/ALTER TABLE.md]: Do we really need user to send a command to convert distribution? I think we can do it when we upgrade.

File: fe/src/main/java/org/apache/doris/catalog/OlapTable.java [hunk 2]
```diff
@@ -1115,4 +1115,20 @@ public int getKeysNum() {
...
+    public boolean convertRandomDistributionToHashDistribution() {
+        boolean hasChanged = false;
+        List<Column> baseSchema = indexIdToSchema.get(baseIndexId);
+        if (defaultDistributionInfo.getType() == DistributionInfoType.RANDOM) {
+            defaultDistributionInfo = ((RandomDistributionInfo) defaultDistributionInfo).toHashDistributionInfo(baseSchema);
+            hasChanged = true;
+        }
+        
+        for (Partition partition : idToPartition.values()) {
+            if (partition.convertRandomDistributionToHashDistribution(baseSchema)) {
```
Comment (imay) [inline fe/src/main/java/org/apache/doris/catalog/OlapTable.java]: In which case defaultDistributionInfo is random, however the partition is not random partition?

Comment (morningman) [reply fe/src/main/java/org/apache/doris/catalog/OlapTable.java]: I don't know, just for safety...

File: docs/documentation/cn/sql-reference/sql-statements/Data Definition/ALTER TABLE.md (hunk 1 above)
Comment (morningman) [reply docs/documentation/cn/sql-reference/sql-statements/Data Definition/ALTER TABLE.md]: I thought about convert it when saving meta image or loading meta image.
But it is very easy to forget to modify someplace, especially when replaying some edit logs related to distribution.

So I think using a command and handle it like a alter table job is more safety.

File: fe/src/main/java/org/apache/doris/analysis/ModifyTablePropertiesClause.java [hunk 3]
```diff
@@ -46,31 +39,24 @@ public void analyze(Analyzer analyzer) throws AnalysisException {
...
-
-            if (!Catalog.getCurrentCatalog().getAuth().checkDbPriv(
-                    ConnectContext.get(), ConnectContext.get().getDatabase(), PrivPredicate.ALTER)) {
-                ErrorReport.reportAnalysisException(ErrorCode.ERR_SPECIFIC_ACCESS_DENIED_ERROR,
-                        "ALTER");
-            }
-        } else if (!Catalog.getCurrentCatalog().getAuth().checkGlobalPriv(ConnectContext.get(), PrivPredicate.ALTER)) {
-            ErrorReport.reportAnalysisException(ErrorCode.ERR_SPECIFIC_ACCESS_DENIED_ERROR,
-                    "ALTER");
+        if (properties.size() != 1) {
```
[the remark this answers is not in the document]
Comment (morningman) [reply fe/src/main/java/org/apache/doris/analysis/ModifyTablePropertiesClause.java]: Privilege is already checked in AlterStmt, here is redundant

[the remark this answers is not in the document]
Comment (imay) [reply fe/src/main/java/org/apache/doris/analysis/ModifyTablePropertiesClause.java]: OK

File: docs/documentation/cn/sql-reference/sql-statements/Data Definition/ALTER TABLE.md (hunk 1 above)
Comment (imay) [reply docs/documentation/cn/sql-reference/sql-statements/Data Definition/ALTER TABLE.md]: ok

File: fe/src/main/java/org/apache/doris/catalog/OlapTable.java (hunk 2 above)
Comment (imay) [reply fe/src/main/java/org/apache/doris/catalog/OlapTable.java]: OK

**Review by imay — APPROVED** [review]
LGTM

**Pull request merged**